src/main/java/core/mate/academy/producer/TrackProducer.java

```diff
@@ -0,0 +1,17 @@
+package core.mate.academy.producer;
+
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackProducer implements MachineProducer {
```

Comment:
Think about why do you need this T in parametrization. Maybe you should use it in the method?
```suggestion
public class TrackProducer implements MachineProducer<T extends Machine> {
```